business/checkers/services/port_mapping_checker.go

```diff
@@ -4,38 +4,50 @@ import (
 	"fmt"
 	"strings"
 
+	apps_v1 "k8s.io/api/apps/v1"
+	core_v1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/intstr"
 
 	"github.com/kiali/kiali/kubernetes"
 	"github.com/kiali/kiali/models"
-	apps_v1 "k8s.io/api/apps/v1"
-	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/labels"
 )
 
 type PortMappingChecker struct {
 	Service     v1.Service
 	Deployments []apps_v1.Deployment
+	Pods        []core_v1.Pod
 }
 
 func (p PortMappingChecker) Check() ([]*models.IstioCheck, bool) {
 	validations := make([]*models.IstioCheck, 0)
 
-	for portIndex, sp := range p.Service.Spec.Ports {
-		if strings.ToLower(string(sp.Protocol)) == "udp" {
-			continue
-		} else if !kubernetes.MatchPortNameWithValidProtocols(sp.Name) {
-			validation := models.Build("port.name.mismatch", fmt.Sprintf("spec/ports[%d]", portIndex))
-			validations = append(validations, &validation)
+	// Check Port naming for services in the service mesh
+	if p.hasMatchingPodsWithSidecar(p.Service) {
+		for portIndex, sp := range p.Service.Spec.Ports {
+			if strings.ToLower(string(sp.Protocol)) == "udp" {
```

Comment:
Not sure if the "udp" is worth to locate it here or inside the "MatchPortNameWithValidProtocols" logic.
But not a blocker, just thinking loud.